lib/rules/no-debug.ts

```diff
@@ -50,7 +51,7 @@ export default createTestingLibraryRule<Options, MessageIds>({
         }
         const initIdentifierNode = getDeepestIdentifierNode(node.init);
 
-        if (!initIdentifierNode) {
+        if (!initIdentifierNode || initIdentifierNode.name === 'console') {
```

Comment:
Shouldn't this actually avoid reporting the discussed case? 🤔

Reply:
Ah, this is not enough. This would just avoid including such a reference in `isDeclaredDebugVariable`. For handling this properly then I'd suggest adding a new list `builtinConsoleNodes` or something like that, then inside `VariableDeclarator` callback if `initIdentifierNode.name === 'console'` you add that node to `builtinConsoleNodes`, and finally before reporting it you need to check `isDebugUtil || isDeclaredDebugVariable || isChainedReferenceDebug` as now AND is not coming from a node within `builtinConsoleNodes`.

I hope that's clear enough 😅

Reply:
It's also valid if just right before reporting the node you get variables references from the node being analyzed with `getVariableReferences` and check if some of those references are `console` to prevent reporting.

Reply:
I have some doubts regarding the approach, but it seems to work so here it is https://github.com/testing-library/eslint-plugin-testing-library/pull/349/commits/268a598efa31af9b2b64f80644891c1f294f887b Thanks for the help!